posthog/src/main/java/com/posthog/internal/PostHogFeatureFlags.kt

```diff
@@ -89,6 +89,17 @@ internal class PostHogFeatureFlags(
 
                 response?.let {
                     synchronized(featureFlagsLock) {
+                        if (response.quotaLimited?.contains("feature_flags") == true) {
+                            config.logger.log("Feature flags are quota limited, clearing existing flags")
+                            this.featureFlags = null
+                            this.featureFlagPayloads = null
+                            config.cachePreferences?.let { preferences ->
+                                preferences.remove(FEATURE_FLAGS)
+                                preferences.remove(FEATURE_FLAGS_PAYLOAD)
+                            }
```

Comment:
what should happen if session replay uses a feature flag and now is quota limited, should it stop working as well? if so, I guess we can also do `preferences.remove(SESSION_REPLAY)` if session replay has a `linkedFlag` value.

Reply:
Oh, yeah, I didn't think of that scenario – can you tell me more about how session replay with flags works? I'm fine removing it but I don't really understand how session replay and flags overlap in this context.

Reply:
well, it should fail closed 

generally speaking folk are setting a linked flag to reduce cost/volume. so failing open would be nasty

so, I have a linked flag, but I am past flag quota, decide/config continues to return the linked flag for replay, the SDK never shows it as received because i'm past quota, and replay never starts

Reply:
then i think the UX is all around how we notify people about being past quota. feels way more important for flags than other products

Reply:
yep, at least failing closed :)

ideally we'd register a super property on events if replay or flags or etc are past their quota for debugging (obvs doesn't help if events are past quota)

but yep, the onward journey will be the thing